File: src/main/java/no/nav/syfo/domain/Tiltak.java
package no.nav.syfo.domain;

import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Accessors(fluent = true)
public class Tiltak {
    public Long id;
    public long oppfoelgingsdialogId;
    public String navn;
    public LocalDate fom;
    public LocalDate tom;
    public String beskrivelse;
    public String status;
    public String gjennomfoering;
    public String beskrivelseIkkeAktuelt;
    public List<Kommentar> kommentarer = new ArrayList<>();

    public LocalDateTime opprettetDato;
    public LocalDateTime sistEndretDato;

    public String opprettetAvAktoerId;
    public String sistEndretAvAktoerId;

}

File: src/main/kotlin/no/nav/syfo/api/v2/domain/oppfolgingsplan/Tiltak.kt
package no.nav.syfo.api.v2.domain.oppfolgingsplan

import java.time.LocalDate
import java.time.LocalDateTime

data class Tiltak(
    val tiltakId: Long,
    val tiltaknavn: String,
    val knyttetTilArbeidsoppgaveId: Long? = null,
    val fom: LocalDate,
    val tom: LocalDate,
    val beskrivelse: String?,
    val beskrivelseIkkeAktuelt: String? = null,

    val opprettetDato: LocalDateTime,
    val sistEndretDato: LocalDateTime,

    val kommentarer: List<Kommentar> = ArrayList(),
    val status: String,
    val gjennomfoering: String,

    val opprettetAv: Person,
    val sistEndretAv: Person
)
